gcloud-java-resourcemanager/src/test/java/com/google/gcloud/resourcemanager/ResourceManagerImplTest.java

```diff
@@ -321,6 +330,60 @@ public void testUndelete() {
   }
 
   @Test
+  public void testGetPolicy() {
+    assertNull(RESOURCE_MANAGER.getPolicy(COMPLETE_PROJECT.projectId()));
+    RESOURCE_MANAGER.create(COMPLETE_PROJECT);
+    RESOURCE_MANAGER.replacePolicy(COMPLETE_PROJECT.projectId(), POLICY);
+    Policy retrieved = RESOURCE_MANAGER.getPolicy(COMPLETE_PROJECT.projectId());
+    assertEquals(POLICY.bindings(), retrieved.bindings());
+    assertNotNull(retrieved.etag());
+    assertEquals(0, retrieved.version().intValue());
+  }
+
+  @Test
+  public void testReplacePolicy() {
+    try {
+      RESOURCE_MANAGER.replacePolicy("nonexistent-project", POLICY);
+      fail("Project doesn't exist.");
+    } catch (ResourceManagerException e) {
+      assertEquals(403, e.code());
+      assertTrue(e.getMessage().endsWith("project was not found."));
+    }
+    RESOURCE_MANAGER.create(PARTIAL_PROJECT);
+    Policy oldPolicy = RESOURCE_MANAGER.getPolicy(PARTIAL_PROJECT.projectId());
+    RESOURCE_MANAGER.replacePolicy(PARTIAL_PROJECT.projectId(), POLICY);
+    try {
+      RESOURCE_MANAGER.replacePolicy(PARTIAL_PROJECT.projectId(), oldPolicy);
+      fail("Policy with an invalid etag didn't cause error.");
+    } catch (ResourceManagerException e) {
+      assertEquals(409, e.code());
+      assertTrue(e.getMessage().contains("Policy etag mismatch"));
+    }
+    String originalEtag = RESOURCE_MANAGER.getPolicy(PARTIAL_PROJECT.projectId()).etag();
+    Policy newPolicy = RESOURCE_MANAGER.replacePolicy(PARTIAL_PROJECT.projectId(), POLICY);
+    assertEquals(POLICY.bindings(), newPolicy.bindings());
+    assertNotNull(newPolicy.etag());
+    assertNotEquals(originalEtag, newPolicy.etag());
+  }
+
+  @Test
+  public void testTestPermissions() {
+    List<Permission> permissions = ImmutableList.of(Permission.GET);
+    try {
+      RESOURCE_MANAGER.testPermissions("nonexistent-project", permissions);
+    } catch (ResourceManagerException e) {
+      assertEquals(403, e.code());
+      assertEquals("Project nonexistent-project not found.", e.getMessage());
+    }
+    RESOURCE_MANAGER.create(PARTIAL_PROJECT);
+    assertEquals(ImmutableList.of(true),
+        RESOURCE_MANAGER.testPermissions(PARTIAL_PROJECT.projectId(), permissions));
+    assertEquals(ImmutableList.of(true, true),
+        RESOURCE_MANAGER.testPermissions(
```

Comment:
Is this an ideal wrapping? I would move line 382 to 381 and break this assert over two lines only.